crates/hir-ty/src/infer/expr.rs

```diff
@@ -237,6 +237,17 @@ impl<'a> InferenceContext<'a> {
                     Some(type_ref) => self.make_ty(type_ref),
                     None => self.table.new_type_var(),
                 };
+                let ret_ty = if *is_async {
+                    // Use the first type parameter as the output type of future.
+                    // existential type AsyncBlockImplTrait<InnerType>: Future<Output = InnerType>
+                    let impl_trait_id = crate::ImplTraitId::AsyncBlockTypeImplTrait(self.owner, *body);
+                    let opaque_ty_id = self.db.intern_impl_trait_id(impl_trait_id).into();
+                    dbg!(&ret_ty);
+                    TyKind::OpaqueType(opaque_ty_id, Substitution::from1(Interner, ret_ty))
+                        .intern(Interner)
```

Comment:
This was mostly copied from the `Expr::Async` arm on line 152 above.

Reply:
This uses `AsyncBlockTypeImplTrait` but isn't an async block.
Should a new variant be added to `ImplTraitId`? Or maybe `AsyncBlockTypeImplTrait` should be renamed?